### ISSUE TITLE
x11copy.c hangs

### ISSUE THREAD

**Issue by sechshelme** [exebook/x11clipboard#5]
x11copy.c hangs

I wanted to try the demo but the program hangs.
As it stands, case SelectionClear is never executed. So it doesn't get out of the "while" loop.
When I kill the program with Ctrl-C. And then output the clipboard in an editor, the HelloWorld is there.

Only modification I made:
````c
#define hello "Hello World !"
....
// XCopy(selection, (unsigned char*) argv[1], strlen(argv[1]));
XCopy(selection, (unsigned char*) hello, strlen(hello));
```
Tested on a recent Linux Mint

**Comment by exebook** [comment]
It doesn't hang, it waits for some other program to paste. That's how X11 clipboard works.

**Comment by sechshelme** [comment]
True, whom I use in another program, e.g. Codeblooks do something, then something happens.
I included a printf to check.
If I press Ctrl-V in the code block, XNextEvent runs through 3x. And in the Github Edit, even more than 10x, for whatever reason.
Only when I copy a text from somewhere else with Ctrl-C, then your program is terminated via SelectionClear.

```c
	while (1) {
		XNextEvent (display, &event);
		printf("while\n");  // new
		switch (event.type) {
```

If I understand correctly, if I press Ctrl-C under X11, the data is not copied to a clipboard, but only a note is passed to X11?
From this point on, the program must always be composed so that an external program requests the data with Ctrl-V and I can then transfer the data to it. And if I then press Ctrl-C in the external program, my X11 program can exit ready mode?

And if I want to use the whole thing sensibly, all events, e.g. Exposure in the same loop as where SelectionRequest and SelectionsClear is?

But something is still quite strange, when I start your program and it stalls, "Hello World" is still on the clipboard.
